File: python/프로그래머스/숫자카드나누기.py
import math

def solution(arrayA, arrayB):
    answer = 0
    a=arrayA[0]
    b=arrayB[0]

    for i in range(len(arrayA)):
        a=math.gcd(a,arrayA[i])
        b=math.gcd(b,arrayB[i])
    cheA=1
    cheB=1
    for i in range(len(arrayA)):
        if arrayA[i] % b==0:
            cheA=0
        if arrayB[i] % a==0:
            cheB=0

    if cheA==0 and cheB==0:
        return cheA
    else:
        return max(a,b)

File: java/숫자카드나누기.java
class Solution {
    public int solution(int[] arrayA, int[] arrayB) {
        int aa = arrayA[0];
        int bb = arrayB[0];
        
        for (int i = 1; i < arrayA.length; i++) {
            aa = gcd(aa, arrayA[i]);
            bb = gcd(bb, arrayB[i]);
        }
        
        int answer = 0;

        if (isNotDivisible(arrayB, aa)) {
            answer = Math.max(answer, aa);
        }

        if (isNotDivisible(arrayA, bb)) {
            answer = Math.max(answer, bb);
        }

        return answer;
    }

    private boolean isNotDivisible(int[] array, int v) {
        if (v <= 1) return false; 
        for (int num : array) {
            if (num % v == 0) return false;
        }
        return true;
    }

    private int gcd(int a, int b) {
        while (b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }
}
